models/zeroex/optimism/zeroex_api_optimism_fills.sql

```diff
@@ -0,0 +1,365 @@
+            CASE
+                WHEN amount0 < '0' THEN abs(swap.amount0)
+                ELSE abs(swap.amount1) END                                                          AS maker_token_amount_raw,
+            'Uniswap V3 Direct'                                                                     AS type,
+            zeroex_tx.affiliate_address                                                             AS affiliate_address,
+            TRUE                                                                                    AS swap_flag,
+            FALSE                                                                                   AS matcha_limit_order_flag
+    FROM {{ source('uniswap_v3_optimism', 'Pair_evt_Swap') }} swap
+   LEFT JOIN {{ source('uniswap_v3_optimism', 'factory_evt_poolcreated') }} pair ON pair.pool = swap.contract_address
+   JOIN zeroex_tx ON zeroex_tx.tx_hash = swap.evt_tx_hash
+   WHERE sender = '0xdef1abe32c034e558cdd535791643c58a13acc10'
+
+        {% if is_incremental() %}
+        AND swap.evt_block_time >= date_trunc('day', now() - interval '1 week')
+        {% endif %}
+        {% if not is_incremental() %}
+      --  AND swap.evt_block_time >= '{{zeroex_v4_start_date}}'
+        {% endif %}
+
+), */
+all_tx AS (
+    /*
+    SELECT *
+    FROM direct_uniswapv3
+     UNION ALL SELECT *
+    FROM direct_PLP 
+    UNION ALL SELECT *
+    FROM ERC20BridgeTransfer
+    UNION ALL SELECT *
+    FROM BridgeFill
+    UNION ALL */
+    SELECT *
+    FROM NewBridgeFill 
+    UNION ALL SELECT *
+    FROM v4_rfq_fills_no_bridge
+    UNION ALL SELECT *
+    FROM v4_limit_fills_no_bridge
+    UNION ALL SELECT *
+    FROM otc_fills 
+)
+
+SELECT 
+        all_tx.tx_hash,
+        all_tx.evt_index,
+        all_tx.contract_address,
+        all_tx.block_time,
+        try_cast(date_trunc('day', all_tx.block_time) AS date) AS block_date,
+        maker,
+        CASE
+            WHEN taker = '0xdef1abe32c034e558cdd535791643c58a13acc10' THEN tx.from
+            ELSE taker
+        END AS taker, -- fix the user masked by ProxyContract issue
+        taker_token,
+        maker_token,
+        taker_token_amount_raw / pow(10, tp.decimals) AS taker_token_amount,
+        taker_token_amount_raw,
+        maker_token_amount_raw / pow(10, mp.decimals) AS maker_token_amount,
+        maker_token_amount_raw,
+        all_tx.type,
+        affiliate_address,
+        swap_flag,
+        matcha_limit_order_flag,
+        COALESCE((all_tx.maker_token_amount_raw / pow(10, mp.decimals)) * mp.price, (all_tx.taker_token_amount_raw / pow(10, tp.decimals)) * tp.price) AS volume_usd
+FROM all_tx
+INNER JOIN {{ source('optimism', 'transactions')}} tx ON all_tx.tx_hash = tx.hash
+
+{% if is_incremental() %}
+AND tx.block_time >= date_trunc('day', now() - interval '1 week')
+{% endif %}
+{% if not is_incremental() %}
+--AND tx.block_time >= '{{zeroex_v3_start_date}}'
+{% endif %}
```

Comment:
Same as above comment, optimism.transactions is also a large table.
```suggestion
INNER JOIN {{ source('optimism', 'transactions')}} tx ON all_tx.tx_hash = tx.hash

{% if is_incremental() %}
AND tx.block_time >= date_trunc('day', now() - interval '1 week')
{% endif %}
{% if not is_incremental() %}
AND tx.block_time >= '{{zeroex_v3_start_date}}'
{% endif %}
```

Reply:
thanks, updated this